src/connectors/nginx.py

```diff
@@ -0,0 +1,290 @@
+    ("http_user_agent", "VARCHAR(16777216)"),
+    ("http_xff", "VARCHAR(16777216)"),
+    ("http_referer", "VARCHAR(16777216)"),
+    ("http_method", "VARCHAR(16777216)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("redirect_counter", "NUMBER(38,0)"),
+    ("remote_address", "VARCHAR(16777216)"),
+    ("request", "VARCHAR(16777216)"),
+    ("request_time", "FLOAT"),
+    ("requests_length", "NUMBER(38,0)"),
+    ("ssl_session_id", "VARCHAR(16777216)"),
+    ("ssl_session_reused", "VARCHAR(16777216)"),
+    ("status", "NUMBER(38,0)"),
+    ("event_time_other", "VARIANT"),
+    ("n_upstream_attempts", "NUMBER(38,0)"),
+    ("upstream_address", "VARCHAR(16777216)"),
+    ("upstream_response_length", "NUMBER(38,0)"),
+    ("upstream_response_time", "FLOAT"),
+    ("upstream_status", "NUMBER(38,0)"),
+]
+
+ERROR_LANDING_TABLE_COLUMNS = [
+    ("raw", "VARIANT"),
+    ("hash_raw", "NUMBER(19,0)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("log_level", "VARCHAR(16777216)"),
+    ("message", "VARCHAR(16777216)"),
+    ("event_time", "TIMESTAMP_LTZ(9)"),
+    ("pid", "NUMBER(38,0)"),
+    ("tid", "NUMBER(38,0)"),
+]
+
+CONNECT_RESPONSE_MESSAGE = """
+STEP 1: Be sure the Role "{role}" includes the following trust relationship:
+
+{role_trust_relationship}
+
+
+STEP 2: For Role "{role}", ensure the following inline policy:
+
+{role_policy}
+"""
+
+
+def connect(connection_name, options):
+    table_name = f'nginx_{connection_name}_connection'
+    log_landing_table = f'data.{table_name}'
+    error_landing_table = f'data.nginx_{connection_name}_error_connection'
+    prefix = ''
+    bucket_name = ''
+
+    for i in options:
+        if i == '':
+            options.pop(i)
+
+    db.create_table(
+        name=log_landing_table,
+        cols=LOG_LANDING_TABLE_COLUMNS,
+        comment=yaml_dump(module='nginx', **options)
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {log_landing_table} TO ROLE {SA_ROLE}')
+
+    db.create_table(
+        name=error_landing_table,
+        cols=ERROR_LANDING_TABLE_COLUMNS,
+        comment=yaml_dump(module='nginx', **options)
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {error_landing_table} TO ROLE {SA_ROLE}')
+
+    stage = options.get('existing_stage')
+    if stage:
+        stage_name = stage
+    else:
+        stage_name = f'data.nginx_{connection_name}_stage'
+        bucket_name = options['bucket_name']
+        prefix = options['prefix']
+        aws_role = options['aws_role']
+        db.create_stage(
+            name=stage_name,
+            url=f's3://{bucket_name}',
+            prefix=prefix,
+            cloud='aws',
+            credentials=aws_role,
+            file_format=db.TypeOptions(type='JSON')
+        )
+
+    stage_props = db.fetch_props(
+        f'DESC STAGE {stage_name}',
+        filter=('AWS_EXTERNAL_ID', 'SNOWFLAKE_IAM_USER', 'AWS_ROLE', 'URL')
+    )
+
+    if prefix == '':
+        prefix = '/'.join(stage_props['URL'].split('/')[3:-1])
```

Comment:
cleanest would be regex, but cleaner to read if you split out

    url_parts = stage_props['URL'].split('/')

Reply:
Turned this into a regex! you're correct, it's cleaner.